Userland/Libraries/LibGUI/FilePicker.cpp

```diff
@@ -297,8 +304,44 @@ void FilePicker::set_path(const String& path)
     }
 
     auto new_path = LexicalPath(path).string();
-    m_location_textbox->set_icon(FileIconProvider::icon_for_path(new_path).bitmap_for_size(16));
     m_model->set_root_path(new_path);
+
+    m_location_combo->set_icon(FileIconProvider::icon_for_path(new_path).bitmap_for_size(16));
+
+    m_path_history.remove_all_matching([&](String const& v) { return v == m_location_combo->text(); });
+    m_path_history.prepend(m_location_combo->text());
+    save_history();
+}
+
+String FilePicker::history_file_path()
+{
+    return LexicalPath::canonicalized_path(String::formatted("{}/{}", Core::StandardPaths::config_directory(), "FilePickerHistory.txt"));
+}
+
+void FilePicker::load_history()
+{
+    m_path_history.clear();
+    auto file_or_error = Core::File::open(history_file_path(), Core::OpenMode::ReadOnly);
+    if (file_or_error.is_error())
+        return;
```

Comment:
I think this would be nicer if we return `ErrorOr<void>`, use `TRY()` here and `(void)` the result at the call site instead, if we don't care about errors (or at least `dbgln()` it there).